### PR TITLE
runtime: make dialing timeout configurable

### PR DESCRIPTION
**Pull request by snir911**
allow to set dialing timeout in configuration.toml
default is 30s

Fixes: #1789
Signed-off-by: Snir Sheriber <ssheribe@redhat.com>

### CODE REVIEW

File: src/runtime/virtcontainers/pkg/agent/protocols/client/client.go [hunk 1]
```diff
@@ -62,12 +62,17 @@ type dialer func(string, time.Duration) (net.Conn, error)
...
 	grpcAddr, parsedAddr, err := parse(sock)
 	if err != nil {
 		return nil, err
 	}
 
+	if timeout > 0 {
+		agentClientLog.WithField("timeout", timeout).Debug("Default dialing timeout has changed")
+		defaultDialTimeout = time.Duration(timeout) * time.Second
+	}
+
```
Comment (liubin) [inline src/runtime/virtcontainers/pkg/agent/protocols/client/client.go:75]: Nit, Suggest this:

```golang
if timeout <= 0 {
    timeout = defaultDialTimeout
}

conn, err = d(grpcAddr, timeout)
```

Comment (snir911) [reply src/runtime/virtcontainers/pkg/agent/protocols/client/client.go:76]: Hi @liubin the thing is types are different, just pushed another option

**Review by liubin — APPROVED** [review]

File: src/runtime/cli/config/configuration-acrn.toml.in [hunk 2]
```diff
@@ -150,6 +150,10 @@ block_device_driver = "@DEFBLOCKSTORAGEDRIVER_ACRN@"
 
 #debug_console_enabled = true
 
+# Agent connection dialing timeout value in seconds
+# (default: 30)
+# dial_timeout = 30
```
Comment (liubin) [inline src/runtime/cli/config/configuration-acrn.toml.in:155]: I noticed that in the configuration file, config items have no leading spaces after `#`, so you may need to fix like this:
```diff
- # dial_timeout = 30
+ #dial_timeout = 30
```

**Comment by devimc** [comment]
thanks @snir911 , should timeout also configurable through annotations?

**Comment by liubin** [comment]
@snir911 golangci-lint complains 
```
src/runtime/virtcontainers/kata_agent.go:230:16: struct of size 112 bytes could be of size 104 bytes (maligned)
type kataAgent struct {
               ^
```

I tested locally this can fix the check error:

```diff
--- a/src/runtime/virtcontainers/kata_agent.go
+++ b/src/runtime/virtcontainers/kata_agent.go
@@ -237,8 +237,8 @@ type kataAgent struct {
        keepConn       bool
        dynamicTracing bool
        dead           bool
-       kmodules       []string
        dialTimout     uint32
+       kmodules       []string
 
        vmSocket interface{}
        ctx      context.Context

```

**Review by c3d — APPROVED** [review]
Looks good to me.

**Review by devimc — APPROVED** [review]
thanks @snir911

**Comment by c3d** [comment]
ARM CI failing with usual symptom:

```
[preflight] Running pre-flight checks
error execution phase preflight: [preflight] Some fatal errors occurred:
	[ERROR Port-10250]: Port 10250 is in use
```

Commit d7ca640) is supposed to address that I believe. Is it in your branch? 
See #3478

**Comment by fidencio** [comment]
@c3d, the `/test` was run before the ARM CI fix was merged into `main`.  I'll trigger it, just for the sake of ensuring we don't have yet another regression on the ARM front.

/retest-arm

**Pull request merged**